### PR TITLE
C2 karimngr c5s10 11

### PR DESCRIPTION
**Pull request by ghost**
#351

@m0hamed  review the scenario 
@AmrARaouf 
@Mirna-Yacout 
@Omar-Hossam

### CODE REVIEW

**Comment by ghost** [comment]
@Omar-Hossam  please review the code
@AmrARaouf  please review the documentation only test are not added yet

File: arability/app/models/gamer.rb [hunk 1]
```diff
@@ -63,6 +63,59 @@ def self.find_for_database_authentication(gamer_conditions)
...
+  # Success:
+  #   returns a list of the keywords (optionally filtered by categories) 
+  #   similar to the search keyword sorted in lexicographical order
+  # Failure:
+  #   returns an empty list if the email had no matches
+  def self.get_similar_emails(email)
+    return [] if email.blank?
+    email.strip!
+    email_list = self.where("gamers.email LIKE ?", "#{email}%")
+
```
Comment (omar-mohyeldin) [inline arability/app/models/gamer.rb]: empty line of no use, please remove

File: arability/app/views/admin/list_admins.html.erb [hunk 2]
```diff
@@ -22,10 +28,23 @@
...
+						              <h3 id="myModalLabel">تأكيد المسح</h3>
+						            </div>
+						            <div class="modal-body">
+						              <p>هل أنت متأكد من مسح المشرف</p><strong><%= item.username %></strong>
+						            </div>
+						            <div class="modal-footer">
+						              <button class="btn" data-dismiss="modal" aria-hidden="true">غلق</button>
+						              <a class="btn btn-warning" href="/admin/remove_admin?id=<%= item.id %>">مسح</a>
+						            </div>
+						          </div>
```
Comment (omar-mohyeldin) [inline arability/app/views/admin/list_admins.html.erb]: adjust previous code's indentation

File: arability/app/views/admin/make_admin.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+						    	</div>
+						  	</div>
+							<% end %>
+							<input type="hidden" value="<%= form_authenticity_token %>" name="authenticity_token">
+							<div class="control-group">
+								<label class="control-label" for="inputkeyword">البريد الالكترونى</label>
+								<div class="controls">
+									<%= text_field_tag("email", "",
+	  											placeholder: "البريد الالكترونى",
+	  											:class => "span4 autocomplete",
```
Comment (omar-mohyeldin) [inline arability/app/views/admin/make_admin.html.erb]: avoid using =>

File: arability/config/routes.rb [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
   scope "/admin" do 
     match "" => "admin#index", :via => [:get]
+    match 'autocomplete' => 'admin#gamer_email_autocomplete'
```
Comment (omar-mohyeldin) [inline arability/config/routes.rb]: use double quotations

**Comment by omar-mohyeldin** [comment]
@karimngr modify your code, and pull master to make pull request mergeable

**Comment by ghost** [comment]
@AmrARaouf  please review tests

**Comment by ghost** [comment]
@Omar-Hossam  reviewe please

File: arability/app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -52,45 +53,155 @@ def list_prizes
...
+  #   this action lists the admins and links to remove them
+  # Params
+  #   order: the order of the list
+  #   page: the current page of the pagination
+  # Success: 
+  #   refreshes the page and displays admins
+  # Failure: 
+  #   none
   def list_admins
     @list = Gamer.where(:admin => true).order(params[:order])
```
Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb]: the => is not yet modified, and this is your last branch, so you have to modify it here

File: arability/app/views/admin/list_admins.html.erb [hunk 6]
```diff
@@ -8,10 +8,16 @@
...
-						</a>
+						عرض المشرفيين
 					</legend>
+					<% if flash[:success] %>
+						<div class="alert alert-success">
+					    	<button type="button" class="close" data-dismiss="alert">&times;</button>
+					    	<div>
+					    		<%= flash[:success] %>
+					    	</div>
+					  	</div>
```
Comment (omar-mohyeldin) [inline arability/app/views/admin/list_admins.html.erb:19]: allign the closing div with its opening div

File: arability/app/views/admin/list_admins.html.erb [hunk 7]
```diff
@@ -22,10 +28,26 @@
 						</thead>
 						<tbody>
 							<% for item in @list %>
+								<div id="myModal<%= item.id %>" class="modal hide fade" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+						            <div class="modal-header">
```
Comment (omar-mohyeldin) [inline arability/app/views/admin/list_admins.html.erb]: code is indented too much compared to the previous line. code in dive should only be indented by 2 spaces. Please adjust rest of code indentation

**Comment by omar-mohyeldin** [comment]
@karimngr please re-modify your code

File: arability/app/models/gamer.rb [hunk 8]
```diff
@@ -64,6 +64,58 @@ def self.find_for_database_authentication(gamer_conditions)
   end
 
   # Description:
+  #   gets emails similar to an email and sorts 
+  #   result by relevance
+  # Author:
```
Comment (AmrARaouf) [inline arability/app/models/gamer.rb]: please modify this documentation

**Comment by AmrARaouf** [comment]
doc verified

**Comment by AmrARaouf** [comment]
please pull master

**Comment by AmrARaouf** [comment]
please redo tests

**Comment by ghost** [comment]
@AmrARaouf  please review

**Comment by AmrARaouf** [comment]
tests verified

**Comment by Mirna-Yacout** [comment]
@karimngr you forgot the redirect arrows

**Comment by ghost** [comment]
@Mirna-Yacout  review

**Comment by Mirna-Yacout** [comment]
verified

**Comment by omar-mohyeldin** [comment]
@karimngr code verified

**Pull request merged**